plugin/LaserGeometryRenderer.h

```diff
@@ -59,12 +55,48 @@ class LaserGeometryRenderer : public GeometryLayerUser
   int64_t computeVersion() const override
     {return (int64_t)getCurrentFrameNumber();}
 
+  void deactivate() override
+  {
+    BaseClass::deactivate();
+    LaserDevice* laserDevice = device.getDevice();
+    if (laserDevice)
+      laserDevice->updateFrame(std::vector<Point>()); // FIXME: I'm not sure to understand why we need this. Side question: does it work to have multiple renderers?
+  }
+/*
+  // Object
+  void variableChangedCallback(Object* variable, Object* changedObject) override
+  {
+    if (!isActive() && isRenderingServiceCurrent())
+    {
+      LaserDevice* laserDevice = device.getDevice();
+      if (laserDevice) {
+        const std::vector<Point> empty;
+        laserDevice->updateFrame(empty);
+      }
+    }
+  }*/
+
   OIL_OBJECT(LaserGeometryRenderer);
 
 protected:
+  LaserDeviceSelector device;
+  LaserGeometryType geometryType;
+  StrictlyPositiveInteger weight;
+  Percentage maskThreshold;
+  OwnedVector<GeometryMask> masks;
+
+private:
+  DummyGeometry dummyGeometryForMasks;
+  GeometryMaskCompositer maskCompositer;
+
+  GeometryMask& getMask()
+    {return maskCompositer.updateAndGetMask(dummyGeometry, masks);}
```

Comment:
What is `dummyGeometry` here? Could you perhaps construct some temporary value here?

Reply:
sorry that didn't compile indeed. I replaced with a better solution that does not involve dummyGeometry / dummyGeometryForMasks.